sgxs-tools/src/sgx_detect/main.rs

```diff
@@ -260,7 +261,21 @@ impl SgxSupport {
             if let Ok(ref aesm) = aesm_service {
                 dev = dev.einittoken_provider(aesm.clone());
             }
-            Ok(Rc::new(RefCell::new(dev.build())))
+            let device = dev.build();
+            let mut path = device.path();
+            let mut mount_paths = vec![];
+            while let Some(p) = path.parent() {
+                mount_paths.push(p.to_str());
+                path = p;
+            }
+            for mi in MountIter::new().unwrap() {
+                if let Ok(m) = mi {
+                    if mount_paths.contains(&m.dest.to_str()) && m.options.contains(&"noexec".to_string()) {
```

Comment:
I think you need to order the results of `MountIter` by directory depth for this to work, or otherwise make sure you're not just matching mount_paths.last()